### ISSUE TITLE
joinVoiceChannel: Error: could not resolve channel

### ISSUE THREAD

**Issue by jaimehrubiks** [hydrabolt/discord.js#443]
Hi, I have a problem when trying to play sound using discord.js
Actually it works perfect, but after some time or after a user has played some sounds, it stops working.
Suddenly it wont play any sound (neither playFile nor playRawStrean) and the user who can't play sounds must leave the voice channel and rejoin.

This is the error:
http://i.imgur.com/ogHYUtg.png

vChannel is get from a client.on("message, ... as: message.sender.voiceChannel
```
mybot.joinVoiceChannel( vChannel , (error,connection) => {
    if (error){
        console.log("Error joining",error);
        //return; // I tried with and without return
    }
    if (youUrl)
        connection.playRawStream(youstream, play);
    else
        connection.playFile('./sonidos/' + fileName, options, play);
});
function play(error, connection){
    if( error ){
        console.log("Error play",error);
        return;
    }
    connection.on('error'   , (error)  => { console.log(error); youstream.destroy(); programVoiceLeave(true);    });
    connection.on('end'     , ()       => { programVoiceLeave();        });
}
```
I tried testing many things but I can't get the root cause of the problem. I am using the #16 indev branch right now.

**Comment by macdja38** [comment]
you may want to consider an else if on the line with if (youURL). If their's an error connecting to the voice Channel it will probably fail to play music in it.

**Comment by GexoXYZ** [comment]
You're using the #16 Indev branch? 
Can you use either stable `npm i --save discord.js` or the **latest** indev ? `npm install hydrabolt/discord.js#indev`

**Comment by jaimehrubiks** [comment]
@macdja38 I think that is not the problem because it works even when youURL has no way to result true.
@GexoXYZ I tried latest indev by reinstalling and cleaning. 
Same problem, after some time, one user looses the ability to play sounds and must rejoin voice channel.

**Comment by nicholastay** [comment]
Sounds like it's related to what's happening in #360 but not just on bot start when there's no cache(?)

**Comment by GexoXYZ** [comment]
Apparently voiceLeave will sometimes give an undefined channel, don't see how this could happen though? (When a user leaves a channel and doesn't join a new one)

```js
	eventVoiceLeave(user) {
		for (let chan of this.channels.getAll("type", "voice")) {
			if (chan.members.has(user)) {
				chan.members.remove(user);
				user.voiceChannel = null;
				return chan;
			}
		}
		return {server: this};
	}
```

**Comment by Programmix** [comment]
Here's what I do at the moment to get around this issue seems to work most of the time, though it isn't foolproof.

`function resolveVoiceChannel(user, server) {
	if (user.voiceChannel) {
		return user.voiceChannel;
	}
	else {
		var voiceChannels = server.channels.filter((channel) => { return channel.type === "voice"; });

		for (var v = 0; v < voiceChannels.length; v++) {
			var voiceChannel = voiceChannels[v];

			if (voiceChannel.members.has("id", user.id))
				return voiceChannel;
		}

		return null; // not found
	}
}`

**Comment by jaimehrubiks** [comment]
@Programmix Thanks for sharing. Although, as you say, it won't help always, I still need to rejoin voice channel some times.

**Comment by jaimehrubiks** [comment]
I'm gonna be testing another (shitty) solution. It is just an idea, and should be perfectioned for multi-server bots, and even per user 'lastChannel' but it might be useful for some one reading this before we get to a fix.
```js
var lastChannel = null;
function resolveVoiceChannel(user, server) {
    
    if ( user.voiceChannel ) {
        return user.voiceChannel;
    }
    server.channels
        .filter(  (channel) => { 
            return channel.type === 'voice'; 
        } )
        .forEach( (channel) => { 
            if ( channel.members.has('id', user.id) ) 
                lastChannel = channel; 
        } );
    return lastChannel;
    
}
```
I have not tested it yet, will do it soon.

**Comment by GexoXYZ** [comment]
Fixed with https://github.com/hydrabolt/discord.js/commit/f49f1ab3c8d0354e5021d7ff8eb721196e96020f :eyes:

**Comment by jaimehrubiks** [comment]
Good, thanks

**Comment by hydrabolt** [comment]
Fixed in [9.0.2](https://github.com/hydrabolt/discord.js/releases/tag/9.0.2)
